### PR TITLE
chore(ci): ci refactor - Part 1 - Generate ci yaml by js

### PR DESCRIPTION
**Pull request by dsherret**
I was looking at #14448 and I think we need to do this in progressive steps to get it merged. Here's part 1, which is just a straight refactor of our yaml file being generated by javascript.

Next steps:
- Incorporate more of #14448 (ex. using composable steps... again, this was just a straight ci yaml to js)
- Have separate build and test workflows.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -460,50 +391,46 @@ jobs:
...
           matrix.job == 'test' &&
           matrix.profile == 'release' &&
           github.repository == 'denoland/deno' &&
           github.ref == 'refs/heads/main' && !startsWith(github.ref, 'refs/tags/')
         env:
           WPT_FYI_USER: deno
-          WPT_FYI_PW: ${{ secrets.WPT_FYI_PW }}
-          GITHUB_TOKEN: ${{ secrets.DENOBOT_PAT }}
-        run: |
+          WPT_FYI_PW: '${{ secrets.WPT_FYI_PW }}'
```
Comment (bartlomieju) [inline .github/workflows/ci.yml:418]: I'm not so familiar with YAML, but in this situation both sides are equivalent? Adding quotes doesn't change how it's evaluated?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -460,50 +391,46 @@ jobs:
...
+          GITHUB_TOKEN: '${{ secrets.DENOBOT_PAT }}'
+        run: |-
           ./target/release/deno run --allow-all --lock=tools/deno.lock.json \
               ./tools/upload_wptfyi.js $(git rev-parse HEAD) --ghstatus
-
       - name: Run benchmarks
-        if: matrix.job == 'bench' && !startsWith(github.ref, 'refs/tags/')
+        if: 'matrix.job == ''bench'' && !startsWith(github.ref, ''refs/tags/'')'
```
Comment (bartlomieju) [inline .github/workflows/ci.yml:424]: This two single quotes looks strange there around `bench`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dsherret) [reply .github/workflows/ci.yml:418]: `'${{ matrix.os }}'`, `name: '${{ matrix.job }} ${{ matrix.profile }} ${{ matrix.os }}'`, etc. seem to work above, so I assume this works too.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dsherret) [reply .github/workflows/ci.yml:424]: Yeah, I'm not sure what I can do with deno_std to get this to be left alone

Comment (dsherret) [reply .github/workflows/ci.yml:424]: The important thing is more the ci.generate.ts because we're not maintaining this file anymore.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Comment by iuioiua** [comment]
Genuine question: what benefit does generating `ci.yml` provide?

**Comment by dsherret** [comment]
@iuioiua YAML is hard to maintain. By generating it instead we can more easily do complex things (ex. do stuff like apply a condition to a group of steps at one time instead of copying and pasting a condition to each step).

**Comment by iuioiua** [comment]
Ah, I see! Is there any reason why `std/yaml` is imported via the URL instead of using this repos std submodule? I assume it's to avoid breaking changes.

**Comment by dsherret** [comment]
I don't think it matters that much. Importing from the url does make it easier to make changes to that file when the std submodule isn't checked out of git.

**Pull request merged**